packages/connection-form/src/components/advanced-options-tabs/csfle-tab/kms-provider-content.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useCallback } from 'react';
+import { Button, css, Icon } from '@mongodb-js/compass-components';
+import type { UpdateConnectionFormField } from '../../../hooks/use-connect-form';
+import type {
+  KMSField,
+  KMSProviderType,
+  KMSProviderName,
+} from '../../../utils/csfle-kms-fields';
+import type { ConnectionFormError } from '../../../utils/validation';
+import type { ConnectionOptions } from 'mongodb-data-service';
+import KMSProviderCard from './kms-provider-card';
+
+const flexContainerStyles = css({
+  display: 'flex',
+  alignItems: 'center',
+});
+
+const pushRightStyles = css({
+  marginLeft: 'auto',
+});
+
+export function getNextKmsProviderName<T extends KMSProviderType>(
+  kmsProviderType: T,
+  currentProviders: string[] = []
+): KMSProviderName<T> {
+  // For name, we are prefixing it with the type of the provider separated by a colon.
+  // This is because the name of the kms provider should always have the type of the provider
+  // and we are not showing it to the user and it is used internally.
+  if (currentProviders.length === 0) {
+    return `${kmsProviderType}:${kmsProviderType}1`;
+  }
+  const currentNums = currentProviders // local:local1
+    .map((name) => name.split(':')[1]?.replace(kmsProviderType, '')) // '1'
```

Comment:
The `[1]` here works because KMS provider names need to follow a `^<type>:[a-zA-Z0-9_]+$` scheme – should we validate that the name part only consists of `[a-zA-Z0-9_]+` to ensure that it doesn't contain a `:`?

Reply:
good point, should we actually force this pattern (`^<type>:[a-zA-Z0-9_]+$`) on the name of the kms? it will also simply extracting secrets?

Reply:
I think it's okay to do that at least through TextInput validation if we can, yeah